src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+ */
+GLPolarQuad::GLPolarQuad(): Quadrature() {
+  _quad_type = GAUSS_LEGENDRE;
+}
+
+
+/**
+ * @brief Set the number of polar angles to initialize.
+ * @param num_polar the number of polar angles (maximum 6)
+ */
+void GLPolarQuad::setNumPolarAngles(const int num_polar) {
+
+  if (num_polar > 12)
+    log_printf(ERROR, "Unable to set the number of polar angles to %d "
+               "for GLPolarQuad (max 12 angles)", num_polar);
+
+  Quadrature::setNumPolarAngles(num_polar);
+}
+
+
+/**
+ * @brief Routine to initialize the polar quadrature.
+ * @details This routine uses the tabulated values for the Gauss-Legendre
+ *          polar angle quadrature, including the sine thetas and weights.
+ */
+void GLPolarQuad::initialize() {
+
+  /* Call parent class initialize routine */
+  Quadrature::initialize();
+
+  /* Allocate temporary arrays for tabulated quadrature values */
+  double* thetas = new double[_num_polar/2*_num_azim/4];
+
+  /* Tabulated values for the sine thetas and weights for the
+   * Leonard polar angle quadrature */
+  if (_num_polar == 2) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = acos(0.5773502691);
+    }
+  }
+  else if (_num_polar == 4) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = acos(0.3399810435);
+      thetas[a*(_num_polar/2)+1] = acos(0.8611363115);
+    }
+  }
+  else if (_num_polar == 6) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = acos(0.2386191860);
+      thetas[a*(_num_polar/2)+1] = acos(0.6612093864);
+      thetas[a*(_num_polar/2)+2] = acos(0.9324695142);
+    }
+  }
+  else if (_num_polar == 8) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = acos(0.1834346424);
+      thetas[a*(_num_polar/2)+1] = acos(0.5255324099);
+      thetas[a*(_num_polar/2)+2] = acos(0.7966664774);
+      thetas[a*(_num_polar/2)+3] = acos(0.9602898564);
+    }
+  }
+  else if (_num_polar == 10) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = acos(0.1488743387);
+      thetas[a*(_num_polar/2)+1] = acos(0.4333953941);
+      thetas[a*(_num_polar/2)+2] = acos(0.6794095682);
+      thetas[a*(_num_polar/2)+3] = acos(0.8650633666);
+      thetas[a*(_num_polar/2)+4] = acos(0.9739065285);
+    }
+  }
+  else if (_num_polar == 12) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = acos(0.1252334085);
+      thetas[a*(_num_polar/2)+1] = acos(0.3678314989);
+      thetas[a*(_num_polar/2)+2] = acos(0.5873179542);
+      thetas[a*(_num_polar/2)+3] = acos(0.7699026741);
+      thetas[a*(_num_polar/2)+4] = acos(0.9041172563);
+      thetas[a*(_num_polar/2)+5] = acos(0.9815606342);
+    }
+  }
+
+  /* Set the arrays of sin thetas and weights */
+  Quadrature::setThetas(thetas, _num_polar/2*_num_azim/4);
+
+  /* Deallocate temporary arrays */
+  delete [] thetas;
+}
+
+
+void GLPolarQuad::precomputeWeights(bool solve_3D) {
```

Comment:
I don't see `solve_3D` used anywhere - is it a stub for something to come later?